src/main/business/UserManager/UserController.js

```diff
@@ -135,12 +133,12 @@ class UserController {
         return {"TAs": businessTAs, "Lecturers": businessLecturers};
     }
 
-    async deleteUser(session, username){
-        await this.verifyType(session, USER_TYPES.ADMIN);
-        if (this.#userRepo.getUser(username).getUserType() === USER_TYPES.ADMIN){
+    async deleteUser(data){
+        this.#verifyType(data.callingUser.type, USER_TYPES.ADMIN);
+        if (this.#userRepo.getUser(data.username).getUserType() === USER_TYPES.ADMIN){
             throw new EMSError("can't delete system admin");
         }
-        this.#userRepo.deleteUser(username);
+        this.#userRepo.deleteUser(data);
```

Comment:
This will surely not work, either fix this by sending the username or refactor the method in the `UserRepo` class